test/realm/function-eval.js

```diff
@@ -174,7 +194,15 @@ test('frozen-eval', t => {
   desc.configurable = false;
   Object.defineProperty(r.global, 'eval', desc);
 
-  t.equal(r.evaluate('(0,eval)(1)'), 1);
+  t.equal(r.evaluate('(0,eval)("1")'), 1);
+  t.equal(
+    r.evaluate(`\
+      const sym = Symbol("foo");
+      (0,eval)(sym) === sym;
+    `),
+    true,
+    'eval(<non-string>) returns <non-string>'
+  );
```

Comment:
So much time has passed, we will probably want a test that verifies the immutability of the `eval` inside a `Compartment`.